pandas/tests/frame/test_block_internals.py

```diff
@@ -162,21 +160,7 @@ def test_constructor_with_convert(self):
         )
         tm.assert_series_equal(result, expected)
 
-    @pytest.mark.xfail(using_string_dtype(), reason="TODO(infer_string)")
     def test_construction_with_mixed(self, float_string_frame, using_infer_string):
-        # test construction edge cases with mixed types
```

Comment:
I assume something got lost in refactor, but the code removed here is not actually being tested. Both the `result` and `expected` variables are getting shadowed before they are compared